validation/phase_test.go

```diff
@@ -0,0 +1,602 @@
+			}
+
+			if test.expectError {
+				require.Error(t, err)
+
+				if test.expectPhaseValidationErr {
+					var phaseErr *PhaseValidationError
+
+					require.ErrorAs(t, err, &phaseErr)
+				}
+			} else {
+				require.NoError(t, err)
+			}
+
+			if !test.useRealValidator && objValidator != nil {
+				objValidator.AssertExpectations(t)
+			}
+		})
+	}
+}
+
+func TestPhaseNameInvalidError(t *testing.T) {
+	t.Parallel()
+
+	err := PhaseNameInvalidError{
+		PhaseName:     "Invalid_Name",
+		ErrorMessages: []string{"contains invalid characters", "too long"},
+	}
+
+	assert.Equal(t, "phase name invalid: contains invalid characters, too long", err.Error())
+}
+
+func TestValidatePhaseName(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name        string
+		phase       types.Phase
+		expectError bool
+	}{
+		{
+			name: "valid phase name",
+			phase: types.Phase{
+				Name: "valid-phase-name",
+			},
+			expectError: false,
+		},
+		{
+			name: "invalid phase name with uppercase",
+			phase: types.Phase{
+				Name: "Invalid-Phase-Name",
+			},
+			expectError: true,
+		},
+		{
+			name: "invalid phase name with underscores",
+			phase: types.Phase{
+				Name: "invalid_phase_name",
+			},
+			expectError: true,
+		},
+		{
+			name: "invalid phase name too long",
+			phase: types.Phase{
+				Name: "this-is-a-very-long-phase-name-that-exceeds-the-dns1035-label-length-limit-of-63-characters",
+			},
+			expectError: true,
+		},
+		{
+			name: "empty phase name",
+			phase: types.Phase{
+				Name: "",
+			},
+			expectError: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			err := validatePhaseName(test.phase)
+
+			if test.expectError {
+				require.Error(t, err)
+
+				var phaseNameErr PhaseNameInvalidError
+
+				assert.ErrorAs(t, err, &phaseNameErr)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestPhaseNameValid(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name      string
+		phaseName string
+		expectErr bool
+	}{
+		{
+			name:      "valid name",
+			phaseName: "valid-phase",
+			expectErr: false,
+		},
+		{
+			name:      "invalid name with uppercase",
+			phaseName: "Invalid",
+			expectErr: true,
+		},
+		{
+			name:      "invalid name with underscores",
+			phaseName: "invalid_name",
+			expectErr: true,
+		},
+		{
+			name:      "empty name",
+			phaseName: "",
+			expectErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			errs := phaseNameValid(test.phaseName)
+
+			if test.expectErr {
+				assert.NotEmpty(t, errs)
+			} else {
+				assert.Empty(t, errs)
+			}
+		})
+	}
+}
+
+func TestPhaseObjectDuplicationError(t *testing.T) {
+	t.Parallel()
+
+	err := PhaseObjectDuplicationError{
+		PhaseNames: []string{"phase1", "phase2", "phase3"},
+	}
+
+	assert.Equal(t, "duplicate object also found in phases phase1, phase2, phase3", err.Error())
```

Comment:
```suggestion
	assert.Equal(t, "duplicate object found in phases phase1, phase2, phase3", err.Error())
```

I'm guessing `phase1`, `phase2`, `phase3` is the full list of phases where this object is present. If so, I'd suggest removing the word "also" since that makes it sound like there is another phase that has this object?